apm-sniffer/apm-agent-core/src/main/java/org/apache/skywalking/apm/agent/core/context/ContextManager.java

```diff
@@ -83,23 +84,23 @@ public static AbstractSpan createEntrySpan(String operationName, ContextCarrier
         AbstractSpan span;
         AbstractTracerContext context;
         operationName = StringUtil.cut(operationName, OPERATION_NAME_THRESHOLD);
-        if (carrier != null && carrier.isValid()) {
+        if (Objects.nonNull(carrier) && carrier.isValid()) {
             SamplingService samplingService = ServiceManager.INSTANCE.findService(SamplingService.class);
             samplingService.forceSampled();
             context = getOrCreate(operationName, true);
-            span = context.createEntrySpan(operationName);
-            context.extract(carrier);
         } else {
             context = getOrCreate(operationName, false);
-            span = context.createEntrySpan(operationName);
         }
-        return span;
+        span = context.createEntrySpan(operationName);
+        context.extract(carrier);
+        return callback(span);
```

Comment:
Why do you still try to change this? You have the tracing context, and `#extract` is in your control already.